parts/linux/cloud-init/artifacts/cse_start.sh

```diff
@@ -49,6 +49,15 @@ JSON_STRING=$( jq -n \
 mkdir -p /var/log/azure/aks
 echo $JSON_STRING | tee /var/log/azure/aks/provision.json
 
+# Create stage marker for two-stage workflow
+if [ "${PRE_PROVISION_ONLY}" = "true" ]; then
+    # Stage 1: Create marker indicating Stage 2 is needed
+    mkdir -p /opt/azure/containers && touch /opt/azure/containers/preprovision.complete
+    echo "Stage 1 complete - kubelet configuration skipped, Stage 2 required" >> /var/log/azure/cluster-provision.log
+    echo "Created preprovision.complete marker file" >> /var/log/azure/cluster-provision.log
+    exit 0
+fi
+# Always create provision.complete to satisfy CSE framework validation
```

Comment:
this comment is misleading, we're only creating provision.complete when PreProvisionOnly = false right?